kbase-extension/static/kbase/js/util/Logging.js

```diff
@@ -0,0 +1,86 @@
+define([], () => {
+    /**
+     * 
+     * @param {string} code - executable Python code in a string
+     * @returns {Promise<string>} - A promise which will result in a string the value
+     *                              of which was that returned by the last statement
+     *                              in the Python code.
+     */
+    async function executePython(code) {
```

Comment:
This seems reasonable. I think there's some other error catching / wait-until-kernel-is-alive code elsewhere that we can either repurpose or just make use of.

Reply:
Ok, I'll hunt for that.